parser_library/src/processing/statement_processors/ordinary_processor.cpp

```diff
@@ -125,6 +132,12 @@ void ordinary_processor::end_processing()
 
     finished_flag_ = true;
 }
+void ordinary_processor::end_statement()
+{
+    listener_.finish_opencode();
+
+    finished_flag_ = true;
```

Comment:
Won't this be called again from `end_processing()` or conversely won't it prevent call to `end_processing()` that does other things as well?

Reply:
I am not sure about this... `lsp_analyzer::opencode_finished()` does something non-trivial and now it will get call twice.